crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -666,7 +666,10 @@ impl SlidingSync {
             }
 
             // Handle the response.
+            // This will batch changes to rooms, so no other service should run at the same time
+            let lock = this.inner.client.locks().sync_service_lock.lock().await;
             let updates = this.handle_response(response, &mut position_guard).await?;
+            drop(lock);
```

Comment:
Could the lock be taken only in this block, instead https://github.com/matrix-org/matrix-rust-sdk/blob/main/crates/matrix-sdk/src/sliding_sync/mod.rs#L328-L349? It should be sufficient.